packages/build/src/plugins/node_version.js

```diff
@@ -44,32 +45,35 @@ const checkForOldNodeVersions = function ({ pluginsOptions, userNodeVersion, log
 
 const addPluginNodeVersion = function ({
   pluginOptions,
+  pluginOptions: { loadedFrom },
   currentNodeVersion,
   userNodeVersion,
   mode,
   nodePath,
   featureFlags,
 }) {
-  // We want to slowly change all the production plugin executions to use our system node version
-  if (featureFlags.buildbot_build_plugins_system_node_version) {
-    return { ...pluginOptions, nodePath: execPath, nodeVersion: currentNodeVersion }
+  if (loadedFrom === 'local' || loadedFrom === 'package.json') {
+    return nonUIPluginNodeVersion({ pluginOptions, currentNodeVersion, userNodeVersion, nodePath, featureFlags })
+  }
+  if (loadedFrom !== 'core' && mode !== 'buildbot') {
+    return { ...pluginOptions, nodePath, nodeVersion: userNodeVersion }
   }
-  return legacyPluginNodeVersion({ pluginOptions, currentNodeVersion, userNodeVersion, mode, nodePath })
+
+  return { ...pluginOptions, nodePath: execPath, nodeVersion: currentNodeVersion }
 }
 
-const legacyPluginNodeVersion = function ({
+const nonUIPluginNodeVersion = function ({
```

Comment:
[sand] Should we merge this function to the parent one?

Reply:
We'll hit the ESlint complexity limitation 😕. IMO it also makes it easier to read. Let me know if you have strong feelings around it 👍

Reply:
Sounds good :+1: